frontend/src/components/GemHistory.vue

```diff
@@ -0,0 +1,239 @@
+        metaboliteCount: matchingModel.metabolite_count,
+        geneCount: matchingModel.gene_count,
+      };
+    },
+  },
+  mounted() {
+    this.setup();
+  },
+  methods: {
+    getMatchingModel(model, version) {
+      const integratedModel = this.integratedModels.find(m => m.short_name === model);
+
+      // Patch versions are not shown, so only major and minor numbers are compared
+      const [major, minor] = integratedModel.version.split('.');
+      const [selectedMajor, selectedMinor] = version.split('.');
+
+      return major === selectedMajor && minor === selectedMinor ? integratedModel : null;
+    },
+    setup() {
+      // Make sure that both the integrated models and SVG
+      // are loaded before setting up the chart.
+      if (
+        !this.setupComplete &&
+        this.integratedModels.length > 0 &&
+        this.$refs.inlineSvg.$el.nodeName === 'svg'
+      ) {
+        this.setupChart();
+        this.setupInteractivity();
+        this.setupComplete = true;
+        return;
+      }
+      setTimeout(this.setup, 50);
+    },
+    setupChart() {
+      const circles = this.$refs.inlineSvg.$el.getElementsByTagName('circle');
+
+      Array.from(circles).forEach(circle => {
+        const { model, version } = { ...circle.dataset };
+        const matchingModel = this.getMatchingModel(model, version);
+        if (matchingModel) {
+          circle.style.fill = '#25543c'; // eslint-disable-line no-param-reassign
+        }
+      });
+    },
+    setupInteractivity() {
+      const circles = this.$refs.inlineSvg.$el.getElementsByTagName('circle');
+
+      Array.from(circles).forEach(circle => {
+        circle.addEventListener('click', e => {
+          this.clearSelection();
+
+          const c = e.target;
+          c.classList.add('selected');
+          this.selectedVersion = {
+            ...c.dataset,
+            externalParentIds: JSON.parse(c.dataset.externalParentIds),
+            element: c,
+          };
+
+          this.selectedVersion.externalParentIds.forEach(({ id }) => {
+            const p = Array.from(circles).find(
+              _c => id === `${_c.dataset.model}-${_c.dataset.version}`
+            );
+            p.classList.add('selected');
+          });
+
+          // timeout is needed to make sure the popover element appears before calculation
+          setTimeout(() => {
+            this.popoverOffset = {
+              top: `${parseFloat(c.getAttribute('cy')) + 30 + this.$refs.wrapper.scrollTop}px`, // below version number
+              left: `${
+                parseFloat(c.getAttribute('cx')) -
+                this.$refs.popover.offsetWidth / 2 -
+                this.$refs.wrapper.scrollLeft
+              }px`,
+            };
+          }, 0);
+        });
+      });
+
+      this.$refs.inlineSvg.$el.addEventListener('click', e => {
+        const el = e.target;
+        if (el.tagName !== 'circle') {
+          this.clearSelection();
+        }
+      });
```

Comment:
This only works for dismissing the pop-up only when click some other place on the SVG. Intuitively I was expecting it to be dismissed when click anywhere. Any thoughts?